src/main/java/com/google/opensesame/servlets/UserServlet.java

```diff
@@ -12,81 +23,101 @@
-            .findAny()
-            .orElse(null);
-    if (result == null) {
-      sendRawTextError(
-          response, HttpServletResponse.SC_BAD_REQUEST, "invalid user or user does not exist");
+    String userGithub = request.getParameter("githubID");
+    Key userKey = KeyFactory.createKey(PersonObject.ENTITY_NAME, userGithub);
+
+    PersonObject userObject;
+
+    try {
+      userObject = toPersonObject(datastore.get(userKey));
+    } catch (Exception e) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Person does not exist in Datastore");
+      userObject = null;
     }
 
-    String jsonPerson = new Gson().toJson(result);
+    String jsonPerson = new Gson().toJson(userObject);
     response.setContentType("application/json;");
     response.getWriter().println(jsonPerson);
   }
 
-  /** Sends an error to the client as raw text instead of the default HTML page. */
-  private void sendRawTextError(HttpServletResponse response, int errorCode, String errorMsg)
-      throws IOException {
-    response.setStatus(errorCode);
-    response.setContentType("text/html;");
-    response.getWriter().println(errorMsg);
+  @Override
+  // Send a user to datastore. Update the current information about the user if one exists.
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String userGitHubID = request.getParameter("gitHubID");
+    if (userGitHubID.isBlank()) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid ID");
+      return;
+    }
+    ArrayList<String> userTags =
+        (ArrayList<String>) Arrays.asList(request.getParameterValues("tags"));
+
+    sendData(new PersonBuilder().gitHubID(userGitHubID).interestTags(userTags).buildPerson());
+  }
+
+  // TODO: Add the person entity creation to the PersonObject/make it it's own class
+  /**
+   * Send a person entity to datastore with information about a given PersonObject @param person.
+   *
+   * @param person a person object
+   */
+  public void storePersonObject(PersonObject person) {
+    Entity personEntity = new Entity(PersonObject.ENTITY_NAME, person.getGitHubID());
+    personEntity.setProperty(PersonObject.GITHUB_ID_FIELD, person.getGitHubID());
+    personEntity.setProperty(PersonObject.TAG_LIST_FIELD, person.getTags());
+    datastore.put(personEntity);
+  }
+
+  // TODO: use function in other servlets.
+  /**
+   * Query datastore for entities. comparisons are done with @param field coming first. For
+   * example @param field EQUAL @param thingToBeComparedTo. Look at
+   * https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/Query.FilterOperator
+   * for more information on the @param operator
+   *
+   * @param EntityName the type of entity to be queried for
+   * @param field the field that is being used to query
+   * @param thingToBeCompareTo the object that will be compared to @param field
```

Comment:
Maybe we can call thingToBeCompareTo `operand`? Haha just trying to suggest something a little more official, since we have `operator` already.

Reply:
Or even just `value` I guess

Reply:
Yeah, that makes a lot more sense
```suggestion
   * @param value the object that will be compared to @param field
```